### PR TITLE
Add axis=columns to cumsum, cummin, cummax and cumprod in DataFrame

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes to add axis=columns to cumsum, cummin, cummax and cumprod in DataFrame.
The approach is similar with https://github.com/databricks/koalas/pull/605 (and therefore this PR tries to deduplicate it).

*This requires Spark 3.0+**

### CODE REVIEW

File: databricks/koalas/tests/test_dataframe.py [hunk 1]
```diff
@@ -1079,27 +1082,79 @@ def test_cummin(self):
         self.assert_eq(pdf.cummin(), kdf.cummin())
         self.assert_eq(pdf.cummin(skipna=False), kdf.cummin(skipna=False))
 
+    @unittest.skipIf(
+        LooseVersion(pyspark_version) < LooseVersion('3.0'),
+        "columns axis is supported in Spark 3.0+.")
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_dataframe.py]: Ah .. I forgot that struct as return type was only added lately as of https://github.com/apache/spark/commit/ddc2052ebd247aa2a8dad34fd5c1cd345fa45118#diff-c1cf83efe5a5b1f2f1f770589988e997 and https://issues.apache.org/jira/browse/SPARK-23836.

I tested all of them against Spark 3.0.

**Comment by HyukjinKwon** [comment]
Tests passed.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -361,23 +361,63 @@ def _reduce_for_stat_function(self, sfun, name, axis=None, numeric_only=False):
...
 
-            @pandas_udf(returnType=as_spark_type(pser.dtype.type))
-            def calculate_columns_axis(*cols):
-                return getattr(pd.concat(cols, axis=1), name)(axis=axis, numeric_only=numeric_only)
+        else:
+            raise ValueError("No axis named %s for object type %s." % (axis, type(axis)))
 
-            df = self._sdf.select(calculate_columns_axis(*self._internal.data_scols).alias("0"))
-            return DataFrame(df)["0"]
+    def _compute_columns_axis(self, name, axis, *args, **kwargs):
```
Comment (ueshin) [inline databricks/koalas/frame.py]: Shall we check the pyspark version and raise an exception if the version is old?

**Comment by HyukjinKwon** [comment]
Yes .. I agree with that part too. It's too early to merge this. Maybe I should avoid to document those and raise different error messages. Let me fix it soon.

**Comment by HyukjinKwon** [comment]
Actually, let me reopen this after Spark 3.0 comes out.

**Pull request closed without merging**